src/server/groups/read.ts

```diff
@@ -1,67 +1,133 @@
+        },
+        select: {
+            id: true,
+            order: true,
+        }
     }))
 }
 
-export async function readMembershipsOfGroup(id: number): Promise<ExpandedMembership[]> {
-    const count = await prismaCall(() => prisma.group.count({
+export async function readGroup(id: number): Promise<Group> {
+    return await prismaCall(() => prisma.group.findUniqueOrThrow({
         where: {
             id,
         },
     }))
+}
 
-    if (count !== 1) throw new ServerError('BAD PARAMETERS', 'Kan ikke lese medlemmer til en gruppe som ikke finnes.')
-
+export async function expandGroup(group: GroupWithIncludes): Promise<ExpandedGroup> {
+    const members = await prisma.membership.count({
+        where: getMembershipFilter('ACTIVE', group.id)
+    })
+    const name = inferGroupName(group)
+    const firstOrder = group.memberships.sort((m1, m2) => m1.order - m2.order)[0]?.order ?? group.order
+    return {
+        ...group,
+        members,
+        firstOrder,
+        name,
+    }
+}
 
-    return await prismaCall(() => prisma.membership.findMany({
+export async function readGroupExpanded(id: number): Promise<ExpandedGroup> {
+    const group = await prismaCall(() => prisma.group.findFirstOrThrow({
         where: {
-            groupId: id,
+            id,
         },
+        include: groupsExpandedIncluder,
     }))
+    return expandGroup(group)
 }
 
-export async function readMembershipsOfGroups(ids: number[]): Promise<ExpandedMembership[]> {
-    return await prismaCall(() => prisma.membership.findMany({
-        where: {
-            groupId: {
-                in: ids,
-            },
-        },
+
+export async function readGroupsExpanded(): Promise<ExpandedGroup[]> {
+    const groups = await prismaCall(() => prisma.group.findMany({
+        include: groupsExpandedIncluder,
     }))
+
+    const groupsExpanded = await Promise.all(groups.map(expandGroup))
+    return groupsExpanded
 }
 
-export async function readMembershipsOfUser(
-    id: number,
-    order?: number
-): Promise<BasicMembership[]> {
-    if (order === undefined) {
-        order = (await readCurrenOmegaOrder()).order
-    }
+/**
+ * Reads expanded groups and sorts them by group type
+ */
+export async function readGroupsStructured(): Promise<GroupsStructured> {
+    const groups = await readGroupsExpanded()
 
-    return await prismaCall(() => prisma.membership.findMany({
-        where: {
-            userId: id,
-            order,
-        },
-        select: {
-            admin: true,
-            groupId: true,
+    const stuctured = groups.reduce((acc, group) => {
+        if (!acc[group.groupType]) {
+            acc[group.groupType] = {
+                name: GroupTypesConfig[group.groupType].name,
+                description: GroupTypesConfig[group.groupType].description,
+                groups: []
+            }
         }
-    }))
+        acc[group.groupType].groups.push(group)
+        return acc
+    }, {} as GroupsStructured)
```

Comment:
If the database is faulty then maybe not all group types are present. Then some bugs may appear where the object returned are missing some keys.

Reply:
True. THe function should be changed to start the object of with all the keys